### PR TITLE
cephadm: create ceph-exporter sock dir if it's not present

### PR DESCRIPTION
**Pull request by adk3798**
Since this is usually /var/run/ceph/ which ends up getting created by other daemons as well, it was common to see ceph-exporter fail to deploy and then deploy fine after once other daemons were down on the host. I don't see any reason we can't just try to make the directory here instead of bailing out.

Fixes: https://tracker.ceph.com/issues/64491wq

NOTE: with the log scraping, this will be effectively tested through teuthology. If this doesn't work we'll get another failure like the one in the linked tracker.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/cephadm/cephadmlib/daemons/ceph.py
Comment (phlogistonjohn) [inline src/cephadm/cephadmlib/daemons/ceph.py:331]: It feels a bit weird to create stuff in a func called validate().
In the smb pr I was trying to add a new place for a service to create dirs and files it needs - see https://github.com/ceph/ceph/pull/55068/commits/0f6a18a09cba35515e100f457e224537ad1e5608 and how it's used by the new servcie.

Would that be more appropriate for this?

Comment (adk3798) [reply src/cephadm/cephadmlib/daemons/ceph.py]: pulled your patches in and rewrote this to use `prepare_data_dir`

**Review by phlogistonjohn — APPROVED** [review]
I didn't review my patches, your new change LGTM.
You may want to get another review from someone other than me. :-)

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-02-27_05:08:20-orch:cephadm-wip-adk-testing-2024-02-26-1600-distro-default-smithi/

not going to go over the failures in the regular detail as we're still cleaning up the `in cluster log` stuff, but it was mostly just things we still need to add to ignorelists and the mds_upgrade_sequence test having issues. Only test that failed due to PRs in the run was https://pulpito.ceph.com/adking-2024-02-27_05:08:20-orch:cephadm-wip-adk-testing-2024-02-26-1600-distro-default-smithi/7574464 so we can't merge the PRs making changes to mgr/nfs but everything else seems to be okay.

**Pull request merged**
